### PR TITLE
[FLINK-2451] [gelly] examples and library cleanup

### PR DESCRIPTION
**Pull request by vasia**
This PR contains a refactoring and cleanup of the gelly exmaples and library methods, as discussed in the mailing list. Tests have been adjusted accordingly.

### CODE REVIEW

**Comment by fhueske** [comment]
Congrats @vasia for the 1000th PR! :fireworks: :tada: :fireworks: 

:smile:

**Comment by vasia** [comment]
ha! I hadn't noticed! That's nice :))

**Comment by tillrohrmann** [comment]
Great :-)

File: flink-staging/flink-gelly/src/test/java/org/apache/flink/graph/test/GatherSumApplyITCase.java [hunk 1]
```diff
@@ -86,26 +72,35 @@ public void testConnectedComponents() throws Exception {
...
-				"6 Infinity\n" +
-				"7 Infinity\n";
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+		Graph<Long, Double, Double> inputGraph = Graph.fromDataSet(
+				SingleSourceShortestPathsData.getDefaultEdgeDataSet(env),
+				new InitMapperSSSP(), env);
+
+        List<Vertex<Long, Double>> result = inputGraph.run(new GSASingleSourceShortestPaths<Long>(1l, 16))
+        		.getVertices().collect();
```
Comment (andralungu) [inline flink-staging/flink-gelly/src/test/java/org/apache/flink/graph/test/GatherSumApplyITCase.java]: Is there a specific reason for which we decided to use `collect()` in the tests? It does not seem to be a consistency issue. The other tests (in flink) are still using  `compareResultsByLinesInMemory()`. Do we gain anything?

**Comment by andralungu** [comment]
Apart from the minor comment, this looks good to merge.

File: flink-staging/flink-gelly/src/test/java/org/apache/flink/graph/test/GatherSumApplyITCase.java (hunk 1 above)
Comment (vasia) [inline flink-staging/flink-gelly/src/test/java/org/apache/flink/graph/test/GatherSumApplyITCase.java]: Yes, we do! The idea is to lazily mitigate the rest of the tests, too. Take a look at FLINK-2032.

**Comment by vasia** [comment]
Thank you for the review @andralungu. I will rebase and merge this.

**Pull request closed without merging**